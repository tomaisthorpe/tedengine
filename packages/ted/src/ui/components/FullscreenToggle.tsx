import styled from 'styled-components';

const Button = styled.div`
  width: 28px;
  height: 26px;
  border: 4px solid rgba(0, 0, 0, 0.2);
  border-radius: 3px;
  transition-property: border-color;
  transition-duration: 0.2s;
  position: relative;
  margin-bottom: 3px;

  &:hover {
    border-color: rgba(255, 255, 255, 0.3);
  }

  &:hover:after {
    content: 'Toggle Fullscreen';
    position: absolute;
    top: 26px;
    right: -4px;
    background: rgba(0, 0, 0, 0.5);
    padding: 3px 6px;
    border-radius: 4px;
    border: 1px solid rgba(0, 0, 0, 0.2);
    font-size: 12px;
    color: white;
    display: flex;
    align-items: center;
    justify-content: center;
    white-space: nowrap;
    cursor: pointer;
  }
`;

export default function FullscreenToggle({
  toggleFullscreen,
}: {
  toggleFullscreen: () => void;
}) {
  return <Button onClick={toggleFullscreen} />;
}
