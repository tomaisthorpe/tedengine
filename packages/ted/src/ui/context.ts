import * as React from 'react';
import type TEventQueue from '../core/event-queue';

export interface TEventQueueContextData {
  events: TEventQueue | undefined;
}

export const TEventQueueContext = React.createContext<TEventQueueContextData>(
  undefined!,
);

export interface TGameContextData {
  [key: string]: any;
}

export const TGameContext = React.createContext<TGameContextData>(undefined!);

export interface TEngineContextData {
  loading: boolean;
}

export const TEngineContext = React.createContext<TEngineContextData>(
  undefined!,
);

export interface TUIContextData {
  renderingSize: {
    width: number;
    height: number;
  };
  scaling: number;
}

export const TUIContext = React.createContext<TUIContextData>(undefined!);
