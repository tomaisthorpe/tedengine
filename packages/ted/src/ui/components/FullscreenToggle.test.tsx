import { render } from '@testing-library/react';
import FullscreenToggle from './FullscreenToggle';
import { TUIContext } from '../context';

test('renders FullscreenToggle component', () => {
  const { container } = render(
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    <TUIContext.Provider
      value={{ scaling: 1, renderingSize: { width: 1, height: 1 } }}
    >
      <FullscreenToggle toggleFullscreen={() => {}} />,
    </TUIContext.Provider>,
  );
  expect(container).toMatchSnapshot();
});
