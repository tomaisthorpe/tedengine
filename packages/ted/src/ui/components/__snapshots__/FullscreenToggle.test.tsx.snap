// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FullscreenToggle component 1`] = `
<div>
  <div
    class="sc-aYaIB gABHKd"
  />
  ,
</div>
`;
